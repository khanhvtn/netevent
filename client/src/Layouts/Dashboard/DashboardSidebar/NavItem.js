import { useLocation, useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Button, ListItem } from '@material-ui/core';
import useStyles from './styles';
import { useDispatch } from 'react-redux';
import { userLogout } from '../../../actions/userActions';

const NavItem = ({ href, icon: Icon, title, ...rest }) => {
    const location = useLocation();
    const dispatch = useDispatch();
    const history = useHistory();
    const css = useStyles();
    const active = href.includes(location.pathname) ? true : false;


    const handleAction = () => {
        if (href === 'logout') {
            return dispatch(userLogout(history));
        }
        if (href === 'pickrole') {
            history.push(`/${href}`)
        }
        history.push(href[0]);
    };

    return (
        <ListItem
            disableGutters
            className={css.sidebarListItem}
            style={{ paddingTop: 0, paddingBottom: 0 }}
            {...rest}
        >
            <Button
                className={css.sidebarListButton}
                color={active ? 'primary' : 'default'}
                onClick={handleAction}
            >
                {Icon && <Icon size="20" />}
                <span>{title}</span>
            </Button>
        </ListItem>
    );
};

NavItem.propTypes = {
    href: PropTypes.string,
    icon: PropTypes.elementType,
    title: PropTypes.string,
};

export default NavItem;