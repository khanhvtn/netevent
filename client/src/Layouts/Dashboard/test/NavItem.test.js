import React from 'react'
import { Provider } from 'react-redux'
import { renderer } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import beforeEach from 'jest-wrap'
import expect from 'expect'
import toJson from 'enzyme-to-json'

import NavItem from '../DashboardSidebar/NavItem'

const mockStore = configureStore([])
describe('NavItem', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            myState: 'Nav Item'
        })

        store.dispatch = jest.fn()

        component = renderer.create(
            <Provider store={store}>
                <NavItem />
            </Provider>
        )
    })
    
    it('should render with given state from Redux store', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})