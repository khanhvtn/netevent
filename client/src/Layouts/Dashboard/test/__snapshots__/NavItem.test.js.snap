// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem should render with given state from Redux store 1`] = `undefined`;
